proto/design/text_style.proto

```diff
@@ -0,0 +1,49 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package com.android.designcompose.proto.design;
+
+import "design/element/font.proto";
+import "design/element/path.proto";
+import "design/modifier/background.proto";
+
+// These are the style properties that apply to text, so we can use them on
+// subsections of a longer string. We then assume that every style transition is
+// a potential line break (and also run the linebreaking algorithm on the
+// content of every style for the normal case where we need to break text that's
+// all in one style).
+message TextStyle {
+  modifier.Background text_color = 1;
+  float font_size = 2;
+  optional string font_family = 3;
+  element.FontWeight font_weight = 4;
```

Comment:
font size and weight can be variables now

Reply:
How was this resolved?

Reply:
Ok. That's been fixed. I just set it to directly use a NumOrVar instead of a seperate message type, and we'll see if we want to change that when we work it into the code